recipe/gen_patch_json.py

```diff
@@ -714,6 +714,19 @@ def _gen_new_index(repodata, subdir):
         if any(dep.startswith("mpich >=3.4.1,") for dep in deps):
             _pin_looser(fn, record, "mpich", upper_bound="4.0")
 
+        # TBB 2021 (oneTBB 2021) is incompatible with previous releases.
+        if has_dep(record, "tbb") and record.get('timestamp', 0) < 1614285451000:
+            for i, dep in enumerate(deps):
+                if dep.split(" ")[0] != "tbb":
+                    continue
+                if "<" in dep:
+                    break
+
+                if " " in record['depends'][i]:
+                    record['depends'][i] = record['depends'][i] + ',<2021a0'
+                else:
+                    _rename_dependency(fn, record, "tbb", "tbb <2021a0")
```

Comment:
this would break if the dependency is `tbb 2020.*` or if it is strictly pinned.
I would suggest the following,
```suggestion
                if dep == "tbb" or any(dep.startswith(f"tbb >={i}") for i in range(2016, 2021)):
                    deps.append("tbb <2021.a0")
                    break
```

Reply:
@isuruf, thank you!

I've applied your suggestion and updated it a bit.
Could you please confirm that it will be fine if a package has 2 "tbb"s under "depends": the first with lower bound (`>=`) or with no bound, and another one with upper bound `<2021.a0`?
E.g.
```
-    "tbb >=2019.0"
+    "tbb >=2019.0",
+    "tbb <2021.a0"
```
While one may expect something like that:
`tbb >=2019.0` --> `tbb >=2019.0,<2021.a0`
`tbb` --> `tbb <2021.a0`

Reply:
Also I have some doubts about `.` before `a0` in your suggestion: is it really needed? I'm not familiar with such versioning details but I see regex in this file:
```python
cb_pin_regex = re.compile(r"^>=(?P<lower>\d(\.\d+)*a?),<(?P<upper>\d(\.\d+)*)a0$")
```
that is used in `_pin_stricter` and `_pin_looser` and it assumes `a0` suffix with no dot before it. BTW this regex won't match for tbb bounds anyway because one-digit major version is expected.

Reply:
>  if a package has 2 "tbb"s under "depends

This is fine, the solver will look for a solution that matches all constraints.

(other people are better qualified than me for the other points)